### PR TITLE
Add Robot Operating System 2.0

### PR DESCRIPTION
**Pull request by fkromer**
https://github.com/fkromer/awesome-ros2

The list is about the Robot Operating System 2.0 which is a is a set of software libraries and tools that help you build robot applications.

# By submitting this pull request I confirm I've read and complied with the below requirements.

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

- I have read and understood the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title.<br>For example, `Add Name of List`, not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should:
	- Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- Be added at the bottom of the appropriate category.
- The list I'm submitting complies with these requirements:
	- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
	- It's the result of hard work and the best I could possibly produce.
	- Non-generated Markdown file in a GitHub repo.
	- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
	- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
	- Not a duplicate.
	- Only has awesome items. Awesome lists are curations of the best, not everything.
	- Includes a project logo/illustration whenever possible.
		- Either fullwidth or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
		- The image should link to the project website or any relevant website.
		- The image should be high-DPI. Set it to maximum half the width of the original image.
	- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
	- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
		- Should be placed on the right side of the readme heading.
		- Should link back to this list.
	- Has a Table of Contents section.
		- Should be named `Contents`, not `Table of Contents`.
		- Should be the first section in the list.
		- Should only have one level of sub-lists, preferably none.
	- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
		- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
		- [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
		- If you use a license badge, it should be SVG, not PNG.
	- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
		- The file should be named `contributing.md`. Casing is up to you.
	- Has consistent formatting and proper spelling/grammar.
		- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
		- The description starts with an uppercase character and ends with a period.
		- Drop all the `A` / `An` prefixes in the descriptions.
		- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
	- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.
- Go to the top and read it again.

### CODE REVIEW

**Comment by fkromer** [comment]
@sindresorhus Could we merge the list or did I miss something from the list above?

File: readme.md [hunk 1]
```diff
@@ -103,6 +103,7 @@
 - [RubyMotion](https://github.com/motion-open-source/awesome-rubymotion) - Write cross-platform native apps for iOS, Android, macOS, tvOS, and watchOS in Ruby.
 - [Smart TV](https://github.com/vitalets/awesome-smart-tv) - Create apps for different TV platforms.
 - [GNOME](https://github.com/Kazhnuz/awesome-gnome) - Simple and distraction-free desktop environment for Linux.
+- [ROS2](https://github.com/fkromer/awesome-ros2) - Robot Operating System 2.0
```
Comment (sindresorhus) [inline readme.md]: Can you make the title be `Robot Operating System 2.0` and use the description field to succinctly describe what it is?

**Comment by sindresorhus** [comment]
> Includes a succinct description of the project/theme at the top of the readme. (Example)

:arrow_up: 

---

The description starts with an uppercase character and **ends with a period**.

--

The `Packages` link in `Contents` doesn't work.

---

The top-level list items in the `Packages` section should be sub-headings instead.

---

`build system` => `Build system` in `build system (Linux)` and `build system (ROS2)`

---

I would like to see more than just libraries. Include some blog posts, tutorials, links to community resources (Stack Overflow tag, IRC, chat, forums, communities, Reddit, etc. Example: https://github.com/sindresorhus/awesome-electron#community).

File: readme.md (hunk 1 above)
Comment (fkromer) [reply readme.md]: Done

**Comment by fkromer** [comment]
- [x] The description starts with an uppercase character and ends with a period.
- [x] The Packages link in Contents doesn't work.
- [x] The top-level list items in the Packages section should be sub-headings instead.
- [x] build system => Build system in build system (Linux) and build system (ROS2)
- [ ] I would like to see more than just libraries. Include some blog posts, tutorials, links to community resources (Stack Overflow tag, IRC, chat, forums, communities, Reddit, etc. Example: https://github.com/sindresorhus/awesome-electron#community). **In progress.**
- [ ] Includes a project logo/illustration whenever possible. The image should be high-DPI. Set it to maximum half the width of the original image. **What do you mean with high-DPI?**

**Comment by legodude17** [comment]
> cartographer real-time simultaneous localization and mapping (SLAM) in 2D and 3D across multiple platforms and sensor configurations

Not quite done. 😄

**Comment by tfoote** [comment]
@fkromer You can use the svg version of the logo from here: https://github.com/ros-infrastructure/artwork/blob/master/ros_logo.svg It's vector so it will scale for any DPI display.

**Comment by fkromer** [comment]
@legodude17 Logo has been replaced with high DPI version of official ROS logo (used for ROS1 and ROS2). :wink:

**Comment by sindresorhus** [comment]
Ping, is everything mentioned fixed?

You also need to fix the merge conflict.

**Comment by sindresorhus** [comment]
Also try to fix as many lint issues as possible when running https://github.com/sindresorhus/awesome-lint

**Comment by fkromer** [comment]
@sindresorhus I added `awesome-lint` to the CI script and I'll fix the [`awesome-lint` issues](https://travis-ci.org/fkromer/awesome-ros2/jobs/438502818) the next days. Do you care about [`awesome_bot` issues](https://travis-ci.org/fkromer/awesome-ros2/jobs/438502819) as well?

**Comment by fkromer** [comment]
@sindresorhus With the commit of this [CI job](https://travis-ci.org/fkromer/awesome-ros2/jobs/438586990) all `awesome-lint` issues are fixed:

- GitHub star badges in the links are quite helpful in my opinion. I'd consider all raised issues `List item description contains invalid markdown  awesome/list-item` false positives.
- All `Link to https://vimeo.com/236168591 is dead      no-dead-urls` issues are false positives as well. The hyperlinks are working just fine.

**Comment by fkromer** [comment]
`awesome-lint` [request to allow badges in links](https://github.com/sindresorhus/awesome-lint/issues/49).

**Comment by sindresorhus** [comment]
You didn't correctly resolve the merge conflict. Look at the diff: https://github.com/sindresorhus/awesome/pull/1152/files

**Comment by sindresorhus** [comment]
From Awesome Lint:

>   ✖   98:65   Text "OS X" should be written as "macOS"         awesome/spell-check

**Comment by fkromer** [comment]
> ✖ 98:65 Text "OS X" should be written as "macOS" awesome/spell-check

[Fixed](https://travis-ci.org/fkromer/awesome-ros2/jobs/445014782)

**Comment by fkromer** [comment]
> You didn't correctly resolve the merge conflict. Look at the diff: https://github.com/sindresorhus/awesome/pull/1152/files

"Cleanup" via force push.

**Comment by sindresorhus** [comment]
The stars badges for items should be after the `.`, not before.

**Comment by sindresorhus** [comment]
The items in the `Presentation` section does not end with a `.`

**Comment by sindresorhus** [comment]
> Do you care about awesome_bot issues as well?

No

**Comment by sindresorhus** [comment]
https://github.com/fkromer/awesome-ros2/blob/master/contributing.md should be customized to your list. Many links doesn't work there too.

**Comment by sindresorhus** [comment]
Please also re-enable the issue tracker.

**Comment by fkromer** [comment]
> Please also re-enable the issue tracker.

Done.

**Comment by fkromer** [comment]
> The items in the `Presentation` section does not end with a `.`

Uhm, I don't think presentation slides titles should end with a `.`

**Comment by fkromer** [comment]
> https://github.com/fkromer/awesome-ros2/blob/master/contributing.md should be customized to your list. Many links doesn't work there too.

Can you guide me to a suitable example list? All lists I have looked at so far just do how I did (copy & paste the awesome-list contributing file).

**Comment by fkromer** [comment]
> The stars badges for items should be after the `.`, not before.

I am pretty sure I tried this before but was confused by `awesome-lint` warning `List item description must end with proper punctuation`. If you are fine with this warning I am happy to change it...

**Comment by sindresorhus** [comment]
> > Includes a succinct description of the project/theme at the top of the readme. (Example)

This is still not fixed. This is my third comment about it.

**Comment by sindresorhus** [comment]
> I tried this before but was confused by `awesome-lint` warning `List item description must end with proper punctuation`... an exemplary [list entry with badge](https://github.com/fkromer/awesome-ros2/commit/34c4403ea2f1ffb1601566b369952c1b4c645eac#diff-04c6e90faac2675aa89e2176d2eec7d8R26) and the corresponding [`awesome-lint` warning](https://travis-ci.org/fkromer/awesome-ros2/jobs/438557373#L459). If you are fine with this warning I am happy to change it...

All the descriptions should end with a dot (before the badge obviously). Awesome-lint might not be expecting a badge after the description. We should improve that.

**Comment by sindresorhus** [comment]
These should not be written in uppercase:

```
INTRODUCING THE Robot Security Framework (RSF), A STANDARDIZED METHODOLOGY TO PERFORM SECURITY ASSESSMENTS IN ROBOTICS
TOWARDS AN OPEN STANDARD FOR ASSESSING THE SEVERITY OF ROBOT SECURITY VULNERABILITIES, THE ROBOT VULNERABILITY SCORING SYSTEM (RVSS)
```

**Comment by fkromer** [comment]
> These should not be written in uppercase:

DONE

**Comment by fkromer** [comment]
> All the descriptions should end with a dot (before the badge obviously). Awesome-lint might not be expecting a badge after the description. We should improve that.

DONE

**Comment by fkromer** [comment]
> The stars badges for items should be after the ., not before.

DONE

**Comment by fkromer** [comment]
> The items in the Presentation section does not end with a .

Are you sure presentation titles should end with a `.`? 😉

**Comment by fkromer** [comment]
> Includes a succinct description of the project/theme at the top of the readme. (Example)

DONE

**Comment by sindresorhus** [comment]
Tweet: https://twitter.com/awesome__re/status/1102200492732571648

**Pull request merged**